common/lib/protocol-definitions/src/protocol.ts

```diff
@@ -93,6 +102,12 @@ export interface IDocumentMessage {
 
     // Traces related to the packet.
     traces?: ITrace[];
+
+    /**
+     * The compression algorithm that was used to compress the op.
+     * @experimental Not ready for use
+     */
+    compression?: CompressionAlgorithm;
```

Comment:
I'd say - at protocol level, this needs to be a string, i.e. not hard code only possible values. In other words, adding new compression algorithm should not be protocol breaking. I'd move actual list of compression algorithms to runtime. While storage may want to know and alter content, it's not required, and I think we will not expect that to happen (at least not in relay service)

Reply:
Yeah I wasn't sure what strategy we wanted to go for here. Using a string here definitely gives us more flexibility if we want to add more algorithms in the future without a breaking change. I'll change this to a string property and add the compression strings to the runtime.